core/amber/src/main/scala/edu/uci/ics/texera/web/resource/WorkflowWebsocketResource.scala

```diff
@@ -140,36 +119,14 @@ class WorkflowWebsocketResource {
   }
 
   def resultPagination(session: Session, request: ResultPaginationRequest): Unit = {
-    val paginatedResultEvent = PaginatedResultEvent(
-      sessionResults
-        .getOrElse(session.getId, Map.empty[String, List[ITuple]])
-        .map {
-          case (operatorID, table) =>
-            (
-              operatorID,
-              table
-                .slice(
-                  request.pageSize * (request.pageIndex - 1),
-                  request.pageSize * request.pageIndex
-                )
-                .map(tuple => tuple.asInstanceOf[Tuple].asKeyValuePairJson())
-            )
-        }
-        .map {
-          case (operatorID, objNodes) =>
-            PaginatedOperatorResult(
-              operatorID,
-              objNodes,
-              sessionResults
-                .getOrElse(session.getId, Map.empty[String, List[ITuple]])
-                .getOrElse(operatorID, List.empty[ITuple])
-                .size
-            )
-        }
-        .toList
-    )
-
-    send(session, paginatedResultEvent)
+    val opResultService = sessionResults(session.getId).operatorResults(request.operatorID)
+    // calculate from index (pageIndex starts from 1 instead of 0)
```

Comment:
I see you refer to this as an index from 1. 
In CS index term is from 0.
Maybe it's more intuitive to call it `pageNum` instead of `pageIndex` as otherwise you have to explain that starts from 1 everywhere?

Reply:
it's just one of the ugly counter-intuitive javascript conventions, even if I change it to `pageNum` I still need to explain it everywhere in case people (and myself) forget it. 

Do you know in Javascript, month starts from 0 instead of 1? If you want to get January, you need to specify `0`.